pandas/core/arrays/categorical.py

```diff
@@ -2191,17 +2192,18 @@ def min(self, numeric_only=None, **kwargs):
         min : the minimum of this `Categorical`
         """
         self.check_for_ordered("min")
-        if numeric_only:
-            good = self._codes != -1
-            pointer = self._codes[good].min(**kwargs)
+        good = self._codes != -1
+        if not good.all():
+            if skipna:
+                pointer = self._codes[good].min(**kwargs)
+            else:
+                return np.nan
```

Comment:
this is not correct for i8 types, which should be pd.NaT. how to fix this?

Reply:
We could check the `categories.dtype.na_value` if it exists. But since this is the current behaviour, it's not critical to fix in this PR I think.